src/application/Export.java

```diff
@@ -1,39 +1,148 @@
+
+        try (ZipOutputStream zipOutputStream = new ZipOutputStream(Files.newOutputStream(zipFilePath))) {
+            Path sourceDirPath = Paths.get(sourceDirectoryPath);
+
+            Files.walk(sourceDirPath).filter(path -> !Files.isDirectory(path))
+                    .forEach(path -> {
+                        ZipEntry zipEntry = new ZipEntry(sourceDirPath.relativize(path).toString());
+                        System.out.println(zipEntry.getName());
+                        if(is_in_xml(zipEntry.getName(), type) == true) {
+
+                            try {
+                                zipOutputStream.putNextEntry(zipEntry);
+                                zipOutputStream.write(Files.readAllBytes(path));
+                                zipOutputStream.closeEntry();
+                            } catch (Exception e) {
+                                System.err.println(e);
+                            }
+                        }
+                    });
+        }
+    }
+
+    private boolean is_in_xml(String nazov, String type){
+
+        if(type.equals("all") && (nazov.equals("images.zip") || nazov.equals("sounds.zip") || nazov.endsWith(".xml"))){
+            return true;
+        }
+        Iterator<Lesson> it = xml.iterator();
+        while(it.hasNext()) {
+            Iterator<Group> iter =  it.next().getGroupsInLesson().iterator();
+            while(iter.hasNext()) {
+                Iterator<Item> i = iter.next().getItemsInGroup().iterator();
+                while(i.hasNext()) {
+                    Item fi = i.next();
+                    //System.out.println(nazov+ "    " +fi.getQuestionImg());
+                    if(type.equals("images")) {
+                        if(fi.getAnswerImg()!=null){
+                            if (fi.getAnswerImg().endsWith(nazov)){
+                                return true;
+                            }
+                        }
+                        else if(fi.getQuestionImg()!=null){
+                            if (fi.getQuestionImg().endsWith(nazov)) {
+                                return true;
+                            }
+                        }
+
+                    }
+                    else if(type.equals("sounds")){
+                        if(fi.getAnswerSound()!=null){
+                            if (fi.getAnswerSound().endsWith(nazov)){
+                                return true;
+                            }
+                        }
+                        if(fi.getQuestionSound() != null) {
+                            if (fi.getQuestionSound().endsWith(nazov)) {
+                                return true;
+                            }
+                        }
+                    }
+
+                }
+            }
+        }
+       return false;
+    }
+
+
 
+    public Boolean hasWriteAccess() {
+        File file = new File(targetPath+"/data.zip");
+        if(file.exists()){
```

Comment:
toto nieje uplne, to ze ten file existuje neznamena ze k nemu mas write access, je tam este ejdna metoda na file objekte (myslim ze sa vola .canWrite()) ktora vracia presne to co my potrebujeme :)

Reply:
+ mozes hned uz vratit tie booleany co ti vracaju tie metody, nemusis tam pridavat 'return true' alebo 'return false'.
Staci 'return file.exists() && file.taDruhaMetodaCoCekujePristupy()'